resources/example-msg-templates/default.html.j2

```diff
@@ -0,0 +1,302 @@
+#}
+
+
+{# You can set any mandatory tags here, and they will be formatted/outputted in the message #}
+{% set requiredTags = ['Application','Owner'] %}
+{% set expireDays = 90 %}
+
+{# The macros below format some resource attributes for better presentation #}
+{% macro getTag(resource, tagKey) -%}
+	{% if resource.get('Tags') %}
+		{% for t in resource.get('Tags') %} 	
+			{%  if t.get('Key') == tagKey %}
+				{{ t.get('Value') }}    							
+			{% endif %}
+		{% endfor %}
+	{% endif %}
+{%- endmacro %}
+	
+{% macro columnHeading(columnNames, tableWidth) -%}
+	<table>
+	{% for columnName in columnNames %}
+		<th>{{ columnName }}</th>
+	{% endfor %}
+{%- endmacro %}
+
+{# The macro below creates the table:
+   Formatting can be dependent on the column names that are passed in 
+#}
+{% macro emailData(resources, columnNames) -%}
+	{% for resource in resources %}
+		<tr>
+		{% for columnName in columnNames %}
+			{% if columnName in requiredTags %}
+				<td>{{ getTag(resource,columnName) }}</td>
+			{% elif columnName == 'tag.Name' %}
+				<td>{{ getTag(resource,'Name') }}</td>
+			{% elif columnName == 'InstanceCount' %}
+				<td align="center">{{ resource['Instances'] | length }}</td>		
+                        {% elif columnName == 'AccessKeys:RemainDays' %}
+                          {% set combinedkeys = dict() %}
+                          {% for accesskeydet in resource['c7n:AccessKeys'] %}
+                            {% set accesskeyid = accesskeydet['AccessKeyId'] %}
+                            {% set remaindays = ( expireDays - (accesskeydet['CreateDate'] | get_date_age)) %}
+                            {% set _dummy = combinedkeys.update({ accesskeyid:remaindays }) %}
+                          {% endfor %}
+                          <td>{{ combinedkeys }}</td>
+                        {% elif columnName == 'RoleLastUsed' %}
+                          {% if resource['RoleLastUsed'].get('LastUsedDate') %}
+                            {% set last_used_status = resource['RoleLastUsed'].get('LastUsedDate')  + ' (Last used ' ~ (resource['RoleLastUsed'].get('LastUsedDate')  | get_date_age)  + ' days ago)' %}
+                          {% else %}
+                            {% set last_used_status = "Never used -" + ' (Created ' ~ (resource['CreateDate'] | get_date_age)  + ' days ago)' %}
+                          {% endif %}
+                          <td>{{ last_used_status }}</td>
+			{% elif columnName == 'VolumeConsumedReadWriteOps' %}
+				<td>{{ resource['c7n.metrics']['AWS/EBS.VolumeConsumedReadWriteOps.Maximum'][0]['Maximum'] }}</td>
+			{% elif columnName == 'PublicIp' %}
+				<td>{{ resource['NetworkInterfaces'][0].get('Association')['PublicIp'] }}</td>										
+			{% else %}
+				<td>{{ resource[columnName] }}</td>
+			{% endif %}
+		{% endfor %}
+		</tr>
+	{% endfor %}
+	</table>
+{%- endmacro %}	
+
+{# Main #}
+{% macro createData(columnNames, resources, tableWidth) %}
```

Comment:
I like how you define a consistent top-level macro for laying out resource data :+1: